compiler/optimizer/TreeInterpreter.hpp

```diff
@@ -24,81 +24,66 @@
 
 #include <stdint.h>
 #include "env/TRMemory.hpp"
-#include "infra/List.hpp"
 #include "optimizer/Optimization.hpp"
-#include "optimizer/Operation.hpp"
-#include <stack>
-#include <string>
-
-
-namespace TR { class Block; }
-namespace TR { class OptimizationManager; }
-namespace TR { class TreeTop; }
 
 namespace TR
 {
+class Block;
+class OptimizationManager;
+class TreeTop;
 
+static const char ValueTypes[9][10] = {
+   "NoType",
+   "Int8",
+   "Int16",
+   "Int32",
+   "Int64",
+   "Float",
+   "Double",
+   "Address",
+   "Boolean"
+};
 
 class TreeInterpreter : public TR::Optimization
 {
    public:
-   // aconst - address constant (zero value means NULL)
-   // iconst - integer constant (32-bit signed 2's complement)
-   // lconst - long integer constant (64-bit signed 2's complement)
-   // fconst - float constant (32-bit ieee fp)
-   // dconst - double constant (64-bit ieee fp)
-   // bconst - byte integer constant (8-bit signed 2's complement)
-   // sconst - short integer constant (16-bit signed 2's complement)
-   char VALUETYPE_NAME[9][10] = {
-      "NoType",
-      "Int8",
-      "Int16",
-      "Int32",
-      "Int64",
-      "Float",
-      "Double",
-      "Address",
-      "Boolean"
-   };
-   typedef enum {
+   enum Datatype {
       NoType=0,
       Int8,
       Int16,
       Int32,
       Int64,
       Float,
       Double,
-      Address,
-      Boolean
-   } DATATYPE;
-   typedef union{
-      uintptrj_t  aconst;
+      Address
+   };
+   union Data{
+      uintptr_t  aconst;
       int8_t      bconst;
       int16_t     sconst;
       int32_t     iconst;
       int64_t     lconst;
       float_t     fconst;
       double_t    dconst;
-      bool        boolean;
-   } DATA;
+   };
    
-   typedef struct {
-      DATATYPE type;
-      DATA data;
-   } VALUE;
+   struct Value {
+      enum Datatype type;
+      union Data data;
```

Comment:
In C++, you don't need the `enum`/`union`/`struct` in front of the type name.

Reply:
Done.